Expensify/App
# Open old dot link when offline or there is network error

@sketchydroide @sobitneupane 

### Details
In order to open old dot link we don't require authToken anymore but will include it we can fetch it.

### Fixed Issues

https://github.com/Expensify/App/issues/10625

### Tests / QA
1. Login to new dot expensify
2. Send an old dot link (example: [www.expensify.com/inbox](http://www.expensify.com/inbox)) to someone.
3. Click on the link.
4. Verify that, when online, opening link works and includes authToken in the url params.
5. Verify that, when offline, opening link works and no growl is displayed.


### PR Review Checklist

#### PR Author Checklist

<details>
<summary><h4>PR Reviewer Checklist</h4>

### QA Steps
1. Login to new dot expensify
2. Send an old dot link (example: [www.expensify.com/inbox](http://www.expensify.com/inbox)) to someone.
3. Click on the link.
4. Verify that, when online, opening link works and includes authToken in the url params.
5. Verify that, when offline, opening link works and no growl is displayed.


### Screenshots


#### MacOS - Chrome

https://user-images.githubusercontent.com/11381171/194131721-4f426092-1896-42d9-8c8c-0b8736ab4a32.mov

#### iOS - Native

https://user-images.githubusercontent.com/11381171/194130869-2346b701-f797-4c0a-b970-0f437943b2d6.mov

#### Desktop
https://user-images.githubusercontent.com/11381171/194672514-becf0ffe-89bf-484d-aae7-23049bac1b72.mp4

#### iOS - Safari

https://user-images.githubusercontent.com/11381171/194672568-8f6dd4b9-753f-4d96-af29-a7208e4b538c.mov


https://user-images.githubusercontent.com/11381171/194672590-0cd9cf71-fae8-487b-a655-3ad634d9bc05.mov

#### Android - Native

https://user-images.githubusercontent.com/11381171/194672731-361a1080-3576-4cfa-afe7-12e886b1bcda.mp4


#### Android - Chrome

https://user-images.githubusercontent.com/11381171/194672969-b86fd173-cf4c-4b2b-b05b-5a024a5a733a.mp4

https://user-images.githubusercontent.com/11381171/194672979-4f4dc686-b29a-45f2-9466-a9255d2dc892.mp4
